chemprop/uncertainty/uncertainty_calibrator.py

```diff
@@ -377,36 +405,44 @@ def calibrate(self):
         individual_vars = np.array(
             self.calibration_predictor.get_individual_vars()
         )  # shape(models, data, tasks)
-        targets = np.array(self.calibration_data.targets())
-        errors = uncal_preds - targets
+        targets = np.array(self.calibration_data.targets(), dtype=float)
+        mask = np.array(self.calibration_data.mask(), dtype=bool)
+        self.var_weighting = np.zeros([self.num_models, targets.shape[1]])  # shape(models, tasks)
 
-        def objective(scaler_values: np.ndarray):
-            scaler_values = np.reshape(softmax(scaler_values), [-1, 1, 1])
-            scaled_vars = np.sum(
-                individual_vars * scaler_values, axis=0, keepdims=False
-            )
-            nll = np.log(2 * np.pi * scaled_vars) / 2 + (errors) ** 2 / (
-                2 * scaled_vars
-            )
-            nll = np.sum(nll, axis=0)
-            return nll
+        for i in range(targets.shape[1]):
+            task_mask = mask[:, i]
+            task_targets = targets[task_mask, i]
+            task_preds = uncal_preds[task_mask, i]
+            task_ind_vars = individual_vars[:, task_mask, i]
+            task_errors = task_preds - task_targets
+
+            def objective(scaler_values: np.ndarray):
+                scaler_values = np.reshape(softmax(scaler_values), [-1, 1])  # (models, 1)
+                scaled_vars = np.sum(
+                    task_ind_vars * scaler_values, axis=0, keepdims=False
+                )  # (data, tasks)
+                nll = np.log(2 * np.pi * scaled_vars) / 2 + (task_errors) ** 2 / (
+                    2 * scaled_vars
+                )
+                nll = np.sum(nll)
+                return nll
 
-        initial_guess = np.ones_like(self.num_models)
-        sol = fmin(objective, initial_guess)
-        self.var_weighting = softmax(sol)
+            initial_guess = np.ones(self.num_models)
+            sol = fmin(objective, initial_guess)
+            self.var_weighting[:,i] = softmax(sol)
         if self.regression_calibrator_metric == "stdev":
             self.scaling = 1
         else:  # interval
             self.scaling = erfinv(self.interval_percentile / 100) * np.sqrt(2)
 
     def apply_calibration(self, uncal_predictor: UncertaintyPredictor):
         uncal_preds = np.array(uncal_predictor.get_uncal_preds())
-        uncal_individual_vars = np.array(uncal_predictor.get_individual_vars())
+        uncal_individual_vars = np.array(uncal_predictor.get_individual_vars())  # shape(models, data, tasks)
         weighted_vars = np.sum(
-            uncal_individual_vars * np.reshape(self.var_weighting, [-1, 1, 1]),
+            uncal_individual_vars * np.expand_dims(self.var_weighting, 1),
             axis=0,
             keepdims=False,
-        )
+        )  #shape(data, tasks)
```

Comment:
```suggestion
        )  # shape(data, tasks)
```
That is a very minor change recommendation. In general, the comment about the shape of matrix in this script looks like `# shape(data, tasks)`  or `# (data, tasks)`. It would be better to let them be consistent (Remove or Retain the `shape` annotation).

Reply:
Yes. I've made them consistent.